plugins/modules/ec2_vpc_net.py

```diff
@@ -369,6 +369,40 @@ def wait_for_vpc_attribute(connection, module, vpc_id, attribute, expected_value
         module.fail_json(msg="Failed to wait for {0} to be updated".format(attribute))
 
 
+def wait_for_vpc_ipv6_state(module, connection, vpc_id, ipv6_assoc_state):
+    """
+    If ipv6_assoc_state is True, wait for VPC to be associated with at least one Amazon-provided IPv6 CIDR block.
+    If ipv6_assoc_state is False, wait for VPC to be dissociated from all Amazon-provided IPv6 CIDR blocks.
+    """
+    start_time = time()
+    criteria_match = False
+    while time() < start_time + 300:
```

Comment:
Just wondering if it would be feasible to define a waiter instead? @tremble

Reply:
I'm trying to add support for opensearch module. To support custom endpoint, I discovered I need to fix issues in the `aws_acm` module to support tags (https://github.com/ansible-collections/community.aws/pull/870), which depends on adding new IAM permissions (https://github.com/mattclay/aws-terminator/pull/188). Along the way, I fixed some doc issues and use of deprecated tasks. Then I found out `aws_acm` needs to support certificate requests (https://github.com/ansible-collections/community.aws/pull/869) so I can issue a certificate for the custom `opensearch` endpoint. Also, certificate requests would potentially require adding a new module for managing private CAs in AWS acm, otherwise we cannot test signing certificate requests with private CAs. Then because I'm testing `opensearch` with IPv6, I discovered I need to fix the IPv6 association in the `ec2_vpc_net` module (#631). The `ec2_vpc_net`  was making changes even in check mode.  Also, the `ec2_route_table` wasn't documented for IPv6 so #634 . While adding integration tests for #634, I found out `community.aws` relies on `ansible.netcommons`, but the `ipsubnet` filter has been moved to `ansible.utils`. That's one more problem to fix that I haven't done yet. Then it turns out both `ansible.netcommons` and `ansible.utils` have an issue that can cause up to 2 ^ 128 objects to be created: https://github.com/ansible-collections/ansible.utils/issues/132. There is also a related 2 ^ N loop in `netaddr`: https://github.com/netaddr/netaddr/issues/241.

Along the way, I was thinking of several other problems (such as  `ec2_vpc_net`  not supporting user-defined IPv6 subnets, missing doc). How about opening tracking issues for these issues that are being discovered along the way, otherwise I'm concerned there will be no end in sight.